service/controller/release/resource/app/desired.go

```diff
@@ -70,7 +70,13 @@ func (r *resourceStateGetter) getDesiredComponents(ctx context.Context, cr *rele
 			return nil, microerror.Mask(err)
 		}
 
-		activeReleases = result.Items
+		for _, r := range result.Items {
+			if key.IsDeleted(&r) {
```

Comment:
We can make it simple as
```
			if !key.IsDeleted(&r) {
				activeReleases = append(activeReleases, r)
			} 			
```

Reply:
Yeah I still prefer current approach. It makes diffs prettier if we add more conditions. I also like _early returns_ to the point it's ridiculous.